lxd/device/disk.go

```diff
@@ -533,6 +533,46 @@ func (d *disk) startVM() (*deviceConfig.RunConfig, error) {
 						time.Sleep(50 * time.Millisecond)
 					}
 				}
+
+				// Start virtiofsd as LXD prefers virtio-fs over 9p. The latter will only be used
+				// as a fallback.
+
+				// Create the socket in this directory instead of the devices directory. QEMU will otherwise
+				// fail with "Permission Denied" which is probably caused by qemu being called with --chroot.
+				sockPath := filepath.Join(d.inst.Path(), fmt.Sprintf("%s.sock", d.name))
+				logPath := filepath.Join(d.inst.LogPath(), fmt.Sprintf("disk.%s.log", d.name))
+
+				// Remove old socket if needed.
+				os.Remove(sockPath)
+
+				// Start the virtiofsd process in non-daemon mode.
+				proc, err := subprocess.NewProcess("/usr/lib/qemu/virtiofsd", []string{fmt.Sprintf("--socket-path=%s", sockPath), "-o", fmt.Sprintf("source=%s", srcPath)}, logPath, logPath)
```

Comment:
This won't work for the snap as `virtiofsd` will just be in PATH.

We need to change both this call and the previous call to `virtfs-proxy-helper` to look for the binary in the PATH and fallback to running from /usr/lib/qemu.

This will also sort out #8065